include/ensmallen_bits/delta_bar_delta/update_policies/momentum_delta_bar_delta_update.hpp

```diff
@@ -0,0 +1,177 @@
+   * @param phi Multiplicative decrease factor for step size.
+   * @param momentum The momentum decay hyperparameter.
+   * @param minGain Minimum allowed gain (scaling factor) for any parameter
+   *     (default: 1e-8).
+   */
+  MomentumDeltaBarDeltaUpdate(
+      const double kappa = 0.2,
+      const double phi = 0.8,
+      const double momentum = 0.5,
+      const double minGain = 1e-8) :
+      kappa(kappa),
+      phi(phi),
+      momentum(momentum),
+      minGain(minGain)
+  {
+    /* Do nothing. */
+  }
+
+  //! Access the kappa hyperparameter.
+  double Kappa() const { return kappa; }
+  //! Modify the kappa hyperparameter.
+  double& Kappa() { return kappa; }
+
+  //! Access the phi hyperparameter.
+  double Phi() const { return phi; }
+  //! Modify the phi hyperparameter.
+  double& Phi() { return phi; }
+
+  //! Access the momentum hyperparameter.
+  double Momentum() const { return momentum; }
+  //! Modify the momentum hyperparameter.
+  double& Momentum() { return momentum; }
+
+  //! Access the minGain hyperparameter.
+  double MinGain() const { return minGain; }
+  //! Modify the minGain hyperparameter.
+  double& MinGain() { return minGain; }
+
+  /**
+   * The UpdatePolicyType policy classes must contain an internal 'Policy'
+   * template class with two template arguments: MatType and GradType. This is
+   * instantiated at the start of the optimization, and holds parameters
+   * specific to an individual optimization.
+   */
+  template <typename MatType, typename GradType>
+  class Policy
+  {
+   public:
+    typedef typename MatType::elem_type ElemType;
+
+    /**
+     * This is called by the optimizer method before the start of the iteration
+     * update process.
+     *
+     * @param parent Instantiated parent class.
+     * @param rows Number of rows in the gradient matrix.
+     * @param cols Number of columns in the gradient matrix.
+     */
+    Policy(
+        const MomentumDeltaBarDeltaUpdate& parent,
+        const size_t rows,
+        const size_t cols) :
+        parent(parent),
+        kappa(ElemType(parent.kappa)),
+        phi(ElemType(parent.phi)),
+        momentum(ElemType(parent.momentum)),
+        minGain(ElemType(parent.minGain))
+    {
+      gains.ones(rows, cols);
+      velocity.zeros(rows, cols);
+    }
+
+    /**
+     * Update step for Gradient Descent.
+     *
+     * @param iterate Parameters that minimize the function.
+     * @param stepSize Step size to be used for the given iteration.
+     * @param gradient The gradient matrix.
+     */
+    void Update(MatType& iterate,
+                const double stepSize,
+                const GradType& gradient)
+    {
+      gains += (sign(gradient) != sign(velocity)) * kappa -
+          (sign(gradient) == sign(velocity)) * (1 - phi) % gains;
+      gains.clamp(minGain, arma::Datum<typename MatType::elem_type>::inf);
+
+      velocity = momentum * velocity - (stepSize * gains) % gradient;
```

Comment:
```suggestion
      velocity = momentum * velocity - (ElemType(stepSize) * gains) % gradient;
```

For this to work with FP16, we actually have to cast the `stepSize` from `double` to whatever it needs to be.  This is because `double` cannot be implicitly cast to `std::float16_t` (or whatever `arma::fp16` is typedef'ed as).

I found this issue while testing the PR with FP16 enabled and Bandicoot available too (I just wanted to make sure that worked because our CI for ensmallen doesn't currently do that right).